contracts/crypto-verify/src/contract.rs

```diff
@@ -112,6 +112,9 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<QueryResponse> {
             &r,
             &s,
         )?),
+        QueryMsg::VerifyBls12PairingEquality { p, q, r, msg, dst } => {
```

Comment:
I think this API is operating on two different levels. One the one hand it operates on points, on the other hand this is specific to a signature verification API that only works for pubkeys in G1.

Let's make this signature verification only, which allows for some simplifications:
- the g1 generator does not need to be passed by the caller
- `r` can be renamed to `pubkey`
- `q` can be renamed to `siganture`

After that, we can split this into 

- `QueryMsg::VerifyBls12SignatureG1`
- `QueryMsg::VerifyBls12SignatureG2`

where the group is the one of the public key. This then allows us to write a new test for https://github.com/noislabs/drand-verify/blob/1db2248afac44fc2e5c9c78f896b4412d8679914/src/verify.rs#L475-L485 with `const DOMAIN_HASH_TO_G1: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_";`

Reply:
Okay, refactored that and added a test for the swapped parameters